### PR TITLE
fix: avoid logging CREATE TABLE exception

### PR DESCRIPTION
**Pull request by JensonSung**
Current implement cannot log CREATE TABLE statements, exceptions are generated.
Fix this by setting the `couldAffectSchema` flag to true for CREATE TABLE statements.

### CODE REVIEW

File: src/sqlancer/common/log/SQLLoggableFactory.java [hunk 1]
```diff
@@ -24,7 +24,11 @@ protected Loggable createLoggable(String input, String suffix) {
 
     @Override
     public SQLQueryAdapter getQueryForStateToReproduce(String queryString) {
-        return new SQLQueryAdapter(queryString);
+        if (queryString.contains("CREATE TABLE")) {
```
Comment (oraluben) [inline src/sqlancer/common/log/SQLLoggableFactory.java:27]: IMO changes could take place in the default constructor of `SQLQueryAdapter`, not here. That class takes care of those stuffs, i.e. `sqlancer.common.query.SQLQueryAdapter#checkQueryString`.

File: test/sqlancer/TestLoggableFactory.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import org.junit.jupiter.api.Test;
+import sqlancer.common.log.SQLLoggableFactory;
+
+public class TestLoggableFactory {
+
+    @Test
+    public void testLogCreateTable() {
+        String query = "CREATE TABLE t1 (c1 INT)";
+        SQLLoggableFactory logger = new SQLLoggableFactory();
+        logger.getQueryForStateToReproduce(query);
```
Comment (oraluben) [inline test/sqlancer/TestLoggableFactory.java:12]: Current test didn't check for any oracles and cannot fail, could you add some assertion to test if a create statement is "loggable"? Maybe check `SQLQueryAdapter.couldAffectSchema` field.

**Review by oraluben — COMMENTED** [review]
Thanks for the contribution! The change seems reasonable. I've left some comments that may improve the readability of this change.

File: src/sqlancer/common/log/SQLLoggableFactory.java (hunk 1 above)
Comment (JensonSung) [reply src/sqlancer/common/log/SQLLoggableFactory.java:27]: Agree. Such changes are more reasonable.

File: test/sqlancer/TestLoggableFactory.java (hunk 2 above)
Comment (JensonSung) [reply test/sqlancer/TestLoggableFactory.java]: Agree. I make such changes.

File: src/sqlancer/common/log/SQLLoggableFactory.java (hunk 1 above)
Comment (JensonSung) [reply src/sqlancer/common/log/SQLLoggableFactory.java]: Does the following change confuse other users?
```Java
    public SQLQueryAdapter(String query) {
        this(query, new ExpectedErrors(), query.contains("CREATE TABLE") && !query.startsWith("EXPLAIN"));
    }
```

Comment (oraluben) [reply src/sqlancer/common/log/SQLLoggableFactory.java]: I'd use something like:
```
    public SQLQueryAdapter(String query, ExpectedErrors expectedErrors) {
        this(query, expectedErrors, guessAffectSchemaFromQuery(query));
    }

    private static boolean guessAffectSchemaFromQuery(String query) {
        return query.contains("CREATE TABLE") && !query.startsWith("EXPLAIN");
    }

    private void checkQueryString() {
        if (guessAffectSchemaFromQuery(query) != couldAffectSchema) {
            throw new AssertionError("CREATE TABLE statements should set couldAffectSchema to true");
        }
    }
```

Comment (JensonSung) [reply src/sqlancer/common/log/SQLLoggableFactory.java]: Agree. Thanks for help.
I change the committed code.
PTAL

**Review by oraluben — APPROVED** [review]

File: src/sqlancer/common/query/SQLQueryAdapter.java [hunk 3]
```diff
@@ -46,7 +50,7 @@ private String canonicalizeString(String s) {
     }
 
     private void checkQueryString() {
-        if (query.contains("CREATE TABLE") && !query.startsWith("EXPLAIN") && !couldAffectSchema) {
+        if (guessAffectSchemaFromQuery(query) != couldAffectSchema) {
```
Comment (mrigger) [inline src/sqlancer/common/query/SQLQueryAdapter.java:53]: The comparison here confuses me. I think the pattern before was more readable?

**Comment by mrigger** [comment]
@oraluben, thanks for the ping, I've added one minor comment.

File: src/sqlancer/common/query/SQLQueryAdapter.java (hunk 3 above)
Comment (oraluben) [reply src/sqlancer/common/query/SQLQueryAdapter.java:53]: I move the `query.contains("CREATE TABLE") && !query.startsWith("EXPLAIN")` part to a separate method, for guess `couldAffectSchema` when not given, and to check here.

Comment (mrigger) [reply src/sqlancer/common/query/SQLQueryAdapter.java:53]: I think extracting this part is fine, but the "!=" is confusing.

Comment (JensonSung) [reply src/sqlancer/common/query/SQLQueryAdapter.java:53]: I think only when the flag `couldAffectSchema` is false, we need to check whether such value is proper according to the result of function `guessAffectSchemaFromQuery`. In this case, an exception could occur, such as when setting `couldAffectSchema` to true for ALTER statement. 
The following revision may be acceptable.
```java
    private void checkQueryString() {
        if (!couldAffectSchema) {
            if (guessAffectSchemaFromQuery(query)) {
                throw new AssertionError("CREATE TABLE statements should set couldAffectSchema to true");
            }
        }
    }
```

Comment (mrigger) [reply src/sqlancer/common/query/SQLQueryAdapter.java:53]: Yes, I think that would be easy to understand, so I would prefer this.

Comment (JensonSung) [reply src/sqlancer/common/query/SQLQueryAdapter.java]: Great, I have pushed the changes.
PTAL

**Review by mrigger — APPROVED** [review]

**Pull request merged**